flink-runtime/src/main/java/org/apache/flink/runtime/rpc/akka/AkkaInvocationHandler.java

```diff
@@ -370,4 +375,33 @@ public String getHostname() {
 	public CompletableFuture<Void> getTerminationFuture() {
 		return terminationFuture;
 	}
+
+	static Object deserializeValueIfNeeded(Object o, Method method) {
+		if (o instanceof SerializedValue) {
+			try {
+				return  ((SerializedValue<?>) o).deserializeValue(AkkaInvocationHandler.class.getClassLoader());
+			} catch (IOException | ClassNotFoundException e) {
+				throw new CompletionException(
+					new RpcException(
+						"Could not deserialize the serialized payload of RPC method : " + method.getName(), e));
+			}
+		} else {
+			return o;
+		}
+	}
+
+	static Throwable resolveTimeoutException(Throwable exception, @Nullable Throwable callStackCapture, Method method) {
+		if (callStackCapture == null || (!(exception instanceof akka.pattern.AskTimeoutException))) {
+			return exception;
```

Comment:
The null branch implies that the exception type is dependent on the configuration option, which sounds like an accident waiting to happen if someone wants to handle timeouts explicitly as they would have to know what there are 2 types of timeouts that can occur.

How problematic would it be to always create a TimeoutException, and only have different behaviors for how the stacktrace is set?

Reply:
I think that would be fine.
The main reason that I wanted this configurable is to have a way to turn call stack capture off, because during large RPC storms, this might add significantly to the memory consumption.
But always using a `TimeoutException` and referring to the RPC method name should be uncritical.